HtmlDiff/Diff.cs

```diff
@@ -111,18 +112,26 @@ public static string Execute(string oldText, string newText)
         /// <returns>HTML diff markup</returns>
         public string Build()
         {
-            SplitInputsToWords();
+            //we should check if there is a difference in the first place. we do not want to do all this calculation for nothing.
+            if (_oldText != _newText)
+            {
+                SplitInputsToWords();
+
+                _matchGranularity = Math.Min(MatchGranularityMaximum, Math.Min(_oldWords.Length, _newWords.Length));
 
-            _matchGranularity = Math.Min(MatchGranularityMaximum, Math.Min(_oldWords.Length, _newWords.Length));
+                List<Operation> operations = Operations();
 
-            List<Operation> operations = Operations();
+                foreach (Operation item in operations)
+                {
+                    PerformOperation(item);
+                }
 
-            foreach (Operation item in operations)
+                return _content.ToString();
+            }
+            else 
             {
-                PerformOperation(item);
+                return _newText;
```

Comment:
Thanks for this. It seems like an obvious improvement 👍 

Just a tip, I find it's useful to implement this kind of logic as a [guard clause ](https://sourcemaking.com/refactoring/replace-nested-conditional-with-guard-clauses).

So effectively the change you made would be:

```
if (_oldText == _newText)
{
     return _newText;
}
```